engine.js

```diff
@@ -331,7 +396,65 @@ var hmdajson = require('./lib/hmdajson'),
      * -----------------------------------------------------
      */
 
-    // TODO
+    var resolveArg = function(arg, contextList) {
+        var tokens = arg.split('.');
+        for (var i = 0; i < contextList.length; i++) {
+            var mappedArg = contextList[i];
+
+            for (var j = 0; j < tokens.length; j++) {
+                mappedArg = mappedArg[tokens[j]];
+                if (mappedArg === undefined) {
+                    break;
+                }
+            }
+            
+            if (mappedArg !== undefined) {
+                return mappedArg;
+            }
+        }
+        return arg;
```

Comment:
@doelleri The comparison functions from the macro edits rely on passing in literal values as arguments for their comparison rule. We could refactor this to replace "greater_than" and ".25" as arguments to something like 
```JSON
{
    "property": "result",
    "condition": "greater_than",
    "value": ".25"
}
```
and allow those functions to handle parsing that final comparison.